### PR TITLE
Extract trivial _distri_key helper in OpenQA::Schema::Result::ScheduledProducts

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: lib/OpenQA/Schema/Result/ScheduledProducts.pm [hunk 1]
```diff
@@ -224,6 +227,7 @@ sub _schedule_iso {
     my $onlysame           = delete $args->{_ONLY_OBSOLETE_SAME_BUILD} // 0;
     my $skip_chained_deps  = delete $args->{_SKIP_CHAINED_DEPS}        // 0;
 
+    _handle_distri($args);
```
Comment (kalikiana) [inline lib/OpenQA/Schema/Result/ScheduledProducts.pm:230]: What's this?

Comment (Martchus) [reply lib/OpenQA/Schema/Result/ScheduledProducts.pm:230]: Likely he wanted to do an in-place conversion in the first place.

Comment (okurz) [reply lib/OpenQA/Schema/Result/ScheduledProducts.pm]: oops, sorry about that. This belongs elsewhere :D Fixed

**Review by Martchus — APPROVED** [review]

**Comment by okurz** [comment]
CI failed with

```
Retry 1 of 3 …
[13:24:36] t/25-cache-service.t .. 1/? timeout: sending signal INT to command ‘tools/retry’
```

which is better handled with https://github.com/os-autoinst/openQA/pull/3420 , retrggering the failed test for here.

**Review by kalikiana — APPROVED** [review]

**Pull request merged**